pkg/module/validation.go

```diff
@@ -3,11 +3,100 @@ package module
 import (
 	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
 	"strings"
+	"time"
 
+	"github.com/kyma-project/cli/internal/kube"
+	"go.uber.org/zap"
 	amv "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+const defaultCRName = "default.yaml"
+
+//readDefaultCR reads the default CR file's contents. The first returned value is true if the file exists. It's false if the file does not exist or an error occurred.
+func readDefaultCR(modulePath string) (bool, []byte, error) {
+	//TODO: Do we need to override the name or it's always "default.yaml"?
+	crPath := filepath.Join(modulePath, defaultCRName)
+
+	fileInfo, err := os.Stat(crPath)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil, nil
+	}
+	if err != nil {
+		return false, nil, fmt.Errorf("Error accessing the default CR file `%q`: %w", crPath, err)
+	}
+
+	if !fileInfo.Mode().IsRegular() {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: Not a regular file", crPath)
+	}
+
+	crData, err := os.ReadFile(crPath)
+	if err != nil {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: %w", crPath, err)
+	}
+
+	return true, crData, nil
+}
+
+func ValidateDefaultCR(modulePath string, log *zap.SugaredLogger) (skipped bool, err error) {
+
+	exists, crData, err := readDefaultCR(modulePath)
+	if err != nil {
+		return false, err
+	}
+	if !exists {
+		//If there's no "default.yaml" file, skip the validation.
+		return true, nil
+	}
+
+	//TODO: How to get this directory path programatically? What's the contract?
+	crdDirectoryPath := os.Getenv("CRD_DIRECTORY")
```

Comment:
IMO we should make this check smarter and discover the CRD based on the packaged layers instead of injecting it as an environment variable. Otherwise one has to always define the directory, otherwise validation won't work.

Reply:
@jakobmoellersap Thanks for the idea. I thought about source files scanning. But this could also work. There's one problem though. The validation is meant to "fail fast", without even building the module. But maybe we can "delay" it, actually building the module is not that much time/resource consuming.